pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -260,10 +257,8 @@ public void run() {
           long segEndTime = realtimeSegment.getMaxTime();
 
           TimeUnit timeUnit = schema.getTimeFieldSpec().getOutgoingGranularitySpec().getTimeType();
-          Configuration configuration = new PropertyListConfiguration();
-          configuration.setProperty(IndexLoadingConfigMetadata.KEY_OF_LOADING_INVERTED_INDEX, invertedIndexColumns);
-          IndexLoadingConfigMetadata configMetadata = new IndexLoadingConfigMetadata(configuration);
-          IndexSegment segment = Loaders.IndexSegment.load(new File(resourceDir, segmentMetatdaZk.getSegmentName()), mode, configMetadata);
+          IndexSegment segment = Loaders.IndexSegment.load(new File(resourceDir, segmentMetatdaZk.getSegmentName()), mode,
+              realtimeTableDataManager.getIndexLoadingConfigMetadata());
```

Comment:
IndexLoadingConfig has only 3 things - 
1. list of inverted indexes to create/load
2. segment version
3. star tree version
(3) should not matter because there is no star tree (2) is what we want to add. For (1), we need to ensure that is correctly set. From reading of code, it should be.